erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.js

```diff
@@ -138,6 +138,24 @@ frappe.ui.form.on('Subcontracting Order', {
 	}
 });
 
+frappe.ui.form.on('Subcontracting Order Supplied Item', {
+	sourced_by_supplier: function(frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		row.rate = row.amount = 0;
+		let rm_cost = 0;
+		if (frm.doc.items && frm.doc.supplied_items) {
+			frm.doc.items.forEach(item => {
+				rm_cost = frappe.utils.sum(cur_frm.doc.supplied_items.map(a => {
+					return !a.sourced_by_supplier ? a.amount : 0
+				}))
```

Comment:
This will calculate the sum every loop. Just do it once and use that.

Also, what is `a` :rofl: